Change all enums under WebCore/platform/graphics/filters/ to use serializer

mikhailramalho opened the pull request:
#### 99370f48350dfae138b30520042652145a85eff8
<pre>
Change all enums under WebCore/platform/graphics/filters/ to use serializer
<a href="https://bugs.webkit.org/show_bug.cgi?id=264834">https://bugs.webkit.org/show_bug.cgi?id=264834</a>

Reviewed by Chris Dumez.

This patch changes all enums so they don&apos;t use the EnumTraits template
anymore.

Most of the patch is straightforward: it turns the enum into enum classes,
adds an uint8_t type specifier to the enum classes, and adds the enum scope
in the call sites. The only change that doesn&apos;t follow this is the
removal of FEFirst and FELast from FilterFunction::Type, default values
for the start and end of a subset of values in the enum; they are only
ever used in the isFilterEffect() function, so I just replaced them with
the actual values.

* Source/WebCore/platform/graphics/coreimage/FEColorMatrixCoreImageApplier.mm:
(WebCore::FEColorMatrixCoreImageApplier::supportsCoreImageRendering):
(WebCore::FEColorMatrixCoreImageApplier::apply const):
* Source/WebCore/platform/graphics/coreimage/FEComponentTransferCoreImageApplier.mm:
(WebCore::FEComponentTransferCoreImageApplier::supportsCoreImageRendering):
(WebCore::FEComponentTransferCoreImageApplier::apply const):
* Source/WebCore/platform/graphics/cpu/arm/filters/FECompositeNeonArithmeticApplier.cpp:
(WebCore::FECompositeNeonArithmeticApplier::FECompositeNeonArithmeticApplier):
* Source/WebCore/platform/graphics/filters/DistantLightSource.cpp:
(WebCore::DistantLightSource::DistantLightSource):
* Source/WebCore/platform/graphics/filters/DistantLightSource.h:
* Source/WebCore/platform/graphics/filters/FEColorMatrix.cpp:
(WebCore::FEColorMatrix::resultIsAlphaImage const):
(WebCore::FEColorMatrix::supportedFilterRenderingModes const):
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/filters/FEColorMatrix.h:
* Source/WebCore/platform/graphics/filters/FEComponentTransfer.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/filters/FEComponentTransfer.h:
* Source/WebCore/platform/graphics/filters/FEComposite.cpp:
(WebCore::FEComposite::calculateImageRect const):
(WebCore::FEComposite::createSoftwareApplier const):
(WebCore::operator&lt;&lt;):
(WebCore::FEComposite::externalRepresentation const):
* Source/WebCore/platform/graphics/filters/FEComposite.h:
* Source/WebCore/platform/graphics/filters/FEConvolveMatrix.h:
* Source/WebCore/platform/graphics/filters/FEDisplacementMap.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/filters/FEDisplacementMap.h:
* Source/WebCore/platform/graphics/filters/FEMorphology.h:
* Source/WebCore/platform/graphics/filters/FETurbulence.h:
* Source/WebCore/platform/graphics/filters/FilterFunction.h:
(WebCore::FilterFunction::isFilterEffect const):
* Source/WebCore/platform/graphics/filters/FilterOperation.h:
* Source/WebCore/platform/graphics/filters/FilterRenderingMode.h:
* Source/WebCore/platform/graphics/filters/LightSource.h:
(): Deleted.
* Source/WebCore/platform/graphics/filters/PointLightSource.cpp:
(WebCore::PointLightSource::PointLightSource):
* Source/WebCore/platform/graphics/filters/PointLightSource.h:
* Source/WebCore/platform/graphics/filters/SpotLightSource.cpp:
(WebCore::SpotLightSource::SpotLightSource):
* Source/WebCore/platform/graphics/filters/SpotLightSource.h:
* Source/WebCore/platform/graphics/filters/software/FEColorMatrixSoftwareApplier.cpp:
(WebCore::FEColorMatrixSoftwareApplier::FEColorMatrixSoftwareApplier):
(WebCore::FEColorMatrixSoftwareApplier::applyPlatformAccelerated const):
(WebCore::FEColorMatrixSoftwareApplier::applyPlatformUnaccelerated const):
(WebCore::FEColorMatrixSoftwareApplier::applyPlatform const):
* Source/WebCore/platform/graphics/filters/software/FEComponentTransferSoftwareApplier.cpp:
(WebCore::FEComponentTransferSoftwareApplier::computeLookupTable):
* Source/WebCore/platform/graphics/filters/software/FECompositeSoftwareApplier.cpp:
(WebCore::FECompositeSoftwareApplier::FECompositeSoftwareApplier):
(WebCore::FECompositeSoftwareApplier::apply const):
* Source/WebCore/platform/graphics/filters/software/FECompositeSoftwareArithmeticApplier.cpp:
(WebCore::FECompositeSoftwareArithmeticApplier::FECompositeSoftwareArithmeticApplier):
* Source/WebCore/platform/graphics/filters/software/FEConvolveMatrixSoftwareApplier.h:
* Source/WebCore/platform/graphics/filters/software/FEDisplacementMapSoftwareApplier.cpp:
(WebCore::FEDisplacementMapSoftwareApplier::FEDisplacementMapSoftwareApplier):
(WebCore::FEDisplacementMapSoftwareApplier::xChannelIndex const):
(WebCore::FEDisplacementMapSoftwareApplier::yChannelIndex const):
* Source/WebCore/platform/graphics/filters/software/FEGaussianBlurSoftwareApplier.h:
* Source/WebCore/platform/graphics/filters/software/FEMorphologySoftwareApplier.h:
* Source/WebCore/platform/graphics/filters/software/FETurbulenceSoftwareApplier.h:
* Source/WebCore/rendering/CSSFilter.cpp:
(WebCore::createBrightnessEffect):
(WebCore::createContrastEffect):
(WebCore::createGrayScaleEffect):
(WebCore::createHueRotateEffect):
(WebCore::createInvertEffect):
(WebCore::createOpacityEffect):
(WebCore::createSaturateEffect):
(WebCore::createSepiaEffect):
* Source/WebCore/svg/SVGComponentTransferFunctionElement.cpp:
(WebCore::SVGComponentTransferFunctionElement::attributeChanged):
* Source/WebCore/svg/SVGComponentTransferFunctionElement.h:
(WebCore::SVGPropertyTraits&lt;ComponentTransferType&gt;::highestEnumValue):
(WebCore::SVGPropertyTraits&lt;ComponentTransferType&gt;::toString):
(WebCore::SVGPropertyTraits&lt;ComponentTransferType&gt;::fromString):
* Source/WebCore/svg/SVGFEColorMatrixElement.cpp:
(WebCore::SVGFEColorMatrixElement::isInvalidValuesLength const):
(WebCore::SVGFEColorMatrixElement::attributeChanged):
(WebCore::SVGFEColorMatrixElement::createFilterEffect const):
* Source/WebCore/svg/SVGFEColorMatrixElement.h:
(WebCore::SVGPropertyTraits&lt;ColorMatrixType&gt;::highestEnumValue):
(WebCore::SVGPropertyTraits&lt;ColorMatrixType&gt;::toString):
(WebCore::SVGPropertyTraits&lt;ColorMatrixType&gt;::fromString):
* Source/WebCore/svg/SVGFECompositeElement.cpp:
(WebCore::SVGFECompositeElement::attributeChanged):
* Source/WebCore/svg/SVGFECompositeElement.h:
(WebCore::SVGIDLEnumLimits&lt;CompositeOperationType&gt;::highestExposedEnumValue):
(WebCore::SVGPropertyTraits&lt;CompositeOperationType&gt;::highestEnumValue):
(WebCore::SVGPropertyTraits&lt;CompositeOperationType&gt;::toString):
(WebCore::SVGPropertyTraits&lt;CompositeOperationType&gt;::fromString):
* Source/WebCore/svg/SVGFEDisplacementMapElement.cpp:
(WebCore::SVGFEDisplacementMapElement::attributeChanged):
* Source/WebCore/svg/SVGFEDisplacementMapElement.h:
(WebCore::SVGPropertyTraits&lt;ChannelSelectorType&gt;::highestEnumValue):
(WebCore::SVGPropertyTraits&lt;ChannelSelectorType&gt;::toString):
(WebCore::SVGPropertyTraits&lt;ChannelSelectorType&gt;::fromString):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/273147@main">https://commits.webkit.org/273147@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ad738d3c579381aa1c31835be62b8ee31ba90eb8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34414 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/13228 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36412 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37098 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31127 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/35499 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15615 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10338 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30126 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34931 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11195 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30689 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9783 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9867 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30742 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38381 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31258 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31058 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35943 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9993 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7869 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33887 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11799 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7917 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10545 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10841 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/svg/SVGComponentTransferFunctionElement.cpp:
```diff
@@ -53,7 +53,7 @@ void SVGComponentTransferFunctionElement::attributeChanged(const QualifiedName&
     case AttributeNames::typeAttr: {
         ComponentTransferType propertyValue = SVGPropertyTraits<ComponentTransferType>::fromString(newValue);
-        if (propertyValue > 0)
+        if (static_cast<int>(propertyValue) > 0)
```
Would be nicer to use `enumToUnderlyingType(propertyValue)`

cdumez commented on Source/WebCore/svg/SVGComponentTransferFunctionElement.h:
```diff
@@ -30,22 +30,22 @@ namespace WebCore {
 template<>
 struct SVGPropertyTraits<ComponentTransferType> {
-    static unsigned highestEnumValue() { return FECOMPONENTTRANSFER_TYPE_GAMMA; }
+    static unsigned highestEnumValue() { return static_cast<unsigned>(ComponentTransferType::FECOMPONENTTRANSFER_TYPE_GAMMA); }
```
Would be nicer to use `enumToUnderlyingType()`

cdumez commented on Source/WebCore/svg/SVGFEColorMatrixElement.cpp:
```diff
@@ -54,17 +54,17 @@ bool SVGFEColorMatrixElement::isInvalidValuesLength() const
     case AttributeNames::typeAttr: {
         auto propertyValue = SVGPropertyTraits<ColorMatrixType>::fromString(newValue);
-        if (propertyValue > 0)
+        if (static_cast<int>(propertyValue) > 0)
```
enumToUnderlyingType

cdumez commented on Source/WebCore/svg/SVGFEColorMatrixElement.h:
```diff
@@ -28,20 +28,20 @@ namespace WebCore {
 template<>
 struct SVGPropertyTraits<ColorMatrixType> {
-    static unsigned highestEnumValue() { return FECOLORMATRIX_TYPE_LUMINANCETOALPHA; }
+    static unsigned highestEnumValue() { return static_cast<unsigned>(ColorMatrixType::FECOLORMATRIX_TYPE_LUMINANCETOALPHA); }
```
enumToUnderlyingType

cdumez commented on Source/WebCore/svg/SVGFECompositeElement.cpp:
```diff
@@ -58,7 +58,7 @@ void SVGFECompositeElement::attributeChanged(const QualifiedName& name, const At
     case AttributeNames::operatorAttr: {
         CompositeOperationType propertyValue = SVGPropertyTraits<CompositeOperationType>::fromString(newValue);
-        if (propertyValue > 0)
+        if (static_cast<int>(propertyValue) > 0)
```
enumToUnderlyingType

cdumez commented on Source/WebCore/svg/SVGFECompositeElement.h:
```diff
@@ -28,30 +28,30 @@
 template<>
 struct SVGPropertyTraits<CompositeOperationType> {
-    static unsigned highestEnumValue() { return FECOMPOSITE_OPERATOR_LIGHTER; }
+    static unsigned highestEnumValue() { return static_cast<unsigned>(CompositeOperationType::FECOMPOSITE_OPERATOR_LIGHTER); }
```
enumToUnderlyingType

cdumez commented on Source/WebCore/svg/SVGFECompositeElement.h:
```diff
@@ -28,30 +28,30 @@
 
 template<>
-inline unsigned SVGIDLEnumLimits<CompositeOperationType>::highestExposedEnumValue() { return FECOMPOSITE_OPERATOR_ARITHMETIC; }
+inline unsigned SVGIDLEnumLimits<CompositeOperationType>::highestExposedEnumValue() { return static_cast<unsigned>(CompositeOperationType::FECOMPOSITE_OPERATOR_ARITHMETIC); }
```
enumToUnderlyingType

cdumez commented on Source/WebCore/svg/SVGFEDisplacementMapElement.cpp:
```diff
@@ -55,13 +55,13 @@ void SVGFEDisplacementMapElement::attributeChanged(const QualifiedName& name, co
     case AttributeNames::xChannelSelectorAttr: {
         auto propertyValue = SVGPropertyTraits<ChannelSelectorType>::fromString(newValue);
-        if (propertyValue > 0)
+        if (static_cast<int>(propertyValue) > 0)
```
enumToUnderlyingType

cdumez commented on Source/WebCore/svg/SVGFEDisplacementMapElement.cpp:
```diff
@@ -55,13 +55,13 @@ void SVGFEDisplacementMapElement::attributeChanged(const QualifiedName& name, co
     case AttributeNames::yChannelSelectorAttr: {
         auto propertyValue = SVGPropertyTraits<ChannelSelectorType>::fromString(newValue);
-        if (propertyValue > 0)
+        if (static_cast<int>(propertyValue) > 0)
```
enumToUnderlyingType

cdumez commented on Source/WebCore/svg/SVGFEDisplacementMapElement.h:
```diff
@@ -27,20 +27,20 @@ namespace WebCore {
 template<>
 struct SVGPropertyTraits<ChannelSelectorType> {
-    static unsigned highestEnumValue() { return CHANNEL_A; }
+    static unsigned highestEnumValue() { return static_cast<unsigned>(ChannelSelectorType::CHANNEL_A); }
```
enumToUnderlyingType

cdumez reviewed (approved):
LGTM with changes.

mikhailramalho committed:
Change all enums under WebCore/platform/graphics/filters/ to use serializer
https://bugs.webkit.org/show_bug.cgi?id=264834

Reviewed by Chris Dumez.

This patch changes all enums so they don't use the EnumTraits template
anymore.

Most of the patch is straightforward: it turns the enum into enum classes,
adds an uint8_t type specifier to the enum classes, and adds the enum scope
in the call sites. The only change that doesn't follow this is the
removal of FEFirst and FELast from FilterFunction::Type, default values
for the start and end of a subset of values in the enum; they are only
ever used in the isFilterEffect() function, so I just replaced them with
the actual values.

* Source/WebCore/platform/graphics/coreimage/FEColorMatrixCoreImageApplier.mm:
(WebCore::FEColorMatrixCoreImageApplier::supportsCoreImageRendering):
(WebCore::FEColorMatrixCoreImageApplier::apply const):
* Source/WebCore/platform/graphics/coreimage/FEComponentTransferCoreImageApplier.mm:
(WebCore::FEComponentTransferCoreImageApplier::supportsCoreImageRendering):
(WebCore::FEComponentTransferCoreImageApplier::apply const):
* Source/WebCore/platform/graphics/cpu/arm/filters/FECompositeNeonArithmeticApplier.cpp:
(WebCore::FECompositeNeonArithmeticApplier::FECompositeNeonArithmeticApplier):
* Source/WebCore/platform/graphics/filters/DistantLightSource.cpp:
(WebCore::DistantLightSource::DistantLightSource):
* Source/WebCore/platform/graphics/filters/DistantLightSource.h:
* Source/WebCore/platform/graphics/filters/FEColorMatrix.cpp:
(WebCore::FEColorMatrix::resultIsAlphaImage const):
(WebCore::FEColorMatrix::supportedFilterRenderingModes const):
(WebCore::operator<<):
* Source/WebCore/platform/graphics/filters/FEColorMatrix.h:
* Source/WebCore/platform/graphics/filters/FEComponentTransfer.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/graphics/filters/FEComponentTransfer.h:
* Source/WebCore/platform/graphics/filters/FEComposite.cpp:
(WebCore::FEComposite::calculateImageRect const):
(WebCore::FEComposite::createSoftwareApplier const):
(WebCore::operator<<):
(WebCore::FEComposite::externalRepresentation const):
* Source/WebCore/platform/graphics/filters/FEComposite.h:
* Source/WebCore/platform/graphics/filters/FEConvolveMatrix.h:
* Source/WebCore/platform/graphics/filters/FEDisplacementMap.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/graphics/filters/FEDisplacementMap.h:
* Source/WebCore/platform/graphics/filters/FEMorphology.h:
* Source/WebCore/platform/graphics/filters/FETurbulence.h:
* Source/WebCore/platform/graphics/filters/FilterFunction.h:
(WebCore::FilterFunction::isFilterEffect const):
* Source/WebCore/platform/graphics/filters/FilterOperation.h:
* Source/WebCore/platform/graphics/filters/FilterRenderingMode.h:
* Source/WebCore/platform/graphics/filters/LightSource.h:
(): Deleted.
* Source/WebCore/platform/graphics/filters/PointLightSource.cpp:
(WebCore::PointLightSource::PointLightSource):
* Source/WebCore/platform/graphics/filters/PointLightSource.h:
* Source/WebCore/platform/graphics/filters/SpotLightSource.cpp:
(WebCore::SpotLightSource::SpotLightSource):
* Source/WebCore/platform/graphics/filters/SpotLightSource.h:
* Source/WebCore/platform/graphics/filters/software/FEColorMatrixSoftwareApplier.cpp:
(WebCore::FEColorMatrixSoftwareApplier::FEColorMatrixSoftwareApplier):
(WebCore::FEColorMatrixSoftwareApplier::applyPlatformAccelerated const):
(WebCore::FEColorMatrixSoftwareApplier::applyPlatformUnaccelerated const):
(WebCore::FEColorMatrixSoftwareApplier::applyPlatform const):
* Source/WebCore/platform/graphics/filters/software/FEComponentTransferSoftwareApplier.cpp:
(WebCore::FEComponentTransferSoftwareApplier::computeLookupTable):
* Source/WebCore/platform/graphics/filters/software/FECompositeSoftwareApplier.cpp:
(WebCore::FECompositeSoftwareApplier::FECompositeSoftwareApplier):
(WebCore::FECompositeSoftwareApplier::apply const):
* Source/WebCore/platform/graphics/filters/software/FECompositeSoftwareArithmeticApplier.cpp:
(WebCore::FECompositeSoftwareArithmeticApplier::FECompositeSoftwareArithmeticApplier):
* Source/WebCore/platform/graphics/filters/software/FEConvolveMatrixSoftwareApplier.h:
* Source/WebCore/platform/graphics/filters/software/FEDisplacementMapSoftwareApplier.cpp:
(WebCore::FEDisplacementMapSoftwareApplier::FEDisplacementMapSoftwareApplier):
(WebCore::FEDisplacementMapSoftwareApplier::xChannelIndex const):
(WebCore::FEDisplacementMapSoftwareApplier::yChannelIndex const):
* Source/WebCore/platform/graphics/filters/software/FEGaussianBlurSoftwareApplier.h:
* Source/WebCore/platform/graphics/filters/software/FEMorphologySoftwareApplier.h:
* Source/WebCore/platform/graphics/filters/software/FETurbulenceSoftwareApplier.h:
* Source/WebCore/rendering/CSSFilter.cpp:
(WebCore::createBrightnessEffect):
(WebCore::createContrastEffect):
(WebCore::createGrayScaleEffect):
(WebCore::createHueRotateEffect):
(WebCore::createInvertEffect):
(WebCore::createOpacityEffect):
(WebCore::createSaturateEffect):
(WebCore::createSepiaEffect):
* Source/WebCore/svg/SVGComponentTransferFunctionElement.cpp:
(WebCore::SVGComponentTransferFunctionElement::attributeChanged):
* Source/WebCore/svg/SVGComponentTransferFunctionElement.h:
(WebCore::SVGPropertyTraits<ComponentTransferType>::highestEnumValue):
(WebCore::SVGPropertyTraits<ComponentTransferType>::toString):
(WebCore::SVGPropertyTraits<ComponentTransferType>::fromString):
* Source/WebCore/svg/SVGFEColorMatrixElement.cpp:
(WebCore::SVGFEColorMatrixElement::isInvalidValuesLength const):
(WebCore::SVGFEColorMatrixElement::attributeChanged):
(WebCore::SVGFEColorMatrixElement::createFilterEffect const):
* Source/WebCore/svg/SVGFEColorMatrixElement.h:
(WebCore::SVGPropertyTraits<ColorMatrixType>::highestEnumValue):
(WebCore::SVGPropertyTraits<ColorMatrixType>::toString):
(WebCore::SVGPropertyTraits<ColorMatrixType>::fromString):
* Source/WebCore/svg/SVGFECompositeElement.cpp:
(WebCore::SVGFECompositeElement::attributeChanged):
* Source/WebCore/svg/SVGFECompositeElement.h:
(WebCore::SVGIDLEnumLimits<CompositeOperationType>::highestExposedEnumValue):
(WebCore::SVGPropertyTraits<CompositeOperationType>::highestEnumValue):
(WebCore::SVGPropertyTraits<CompositeOperationType>::toString):
(WebCore::SVGPropertyTraits<CompositeOperationType>::fromString):
* Source/WebCore/svg/SVGFEDisplacementMapElement.cpp:
(WebCore::SVGFEDisplacementMapElement::attributeChanged):
* Source/WebCore/svg/SVGFEDisplacementMapElement.h:
(WebCore::SVGPropertyTraits<ChannelSelectorType>::highestEnumValue):
(WebCore::SVGPropertyTraits<ChannelSelectorType>::toString):
(WebCore::SVGPropertyTraits<ChannelSelectorType>::fromString):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/273147@main
